_rules/SC2-5-3-name-from-content-included-in-accessible-name.md

```diff
@@ -0,0 +1,176 @@
+
+success_criterion:
+- 2.5.3 # Label in Name
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Anne Thyme Nørregaard
+- Bryn Anderson
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+This rule applies to any HTML or SVG element that:
+* has a [semantic role](#semantic-role) that is a [widget](https://www.w3.org/TR/wai-aria-1.1/#widget_roles) that [supports name from content](https://www.w3.org/TR/wai-aria-1.1/#namefromcontent), and 
+* has [visible](#visible) [text](https://www.w3.org/TR/WCAG21/#dfn-text), and
+* has an [accessible name](#accessible-name), even if the name is the empty string ("").
+
+**Note**: [widget roles](https://www.w3.org/TR/wai-aria-1.1/#widget_roles) that [support name from content](https://www.w3.org/TR/wai-aria-1.1/#namefromcontent) are: `button`, `checkbox`, `gridcell`, `link`, `menuitem`, `menuitemcheckbox`, `menuitemradio`, `option`, `radio`, `searchbox`, `switch`, `tab`, `treeitem`.
+
+### Expectation
+
+The complete [visible](#visible) [text](https://www.w3.org/TR/WCAG21/#dfn-text) of the target element is [included](#included-characters) within its [accessible name](#accessible-name).
+
+## Assumptions
+
+This rule assumes that the [visible](#visible) [text](ttps://www.w3.org/TR/WCAG21/#dfn-text) has the same order as the text in the code, e.g. CSS has not been used to re-arrange pieces of text so that they are presented in a different order.
+
+## Accessibility Support
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+
+- https://www.w3.org/TR/WCAG21/#label-in-name
+
+## Test cases
+
+### Passed
+
+#### Passed example 1
+
+Visible label and accessible name matches when trailing white spaces are removed.
+
+```html
+<div role="link" aria-label="next page ">next page</div>
+```
+
+#### Passed example 2
+
+Character insensitivity between visible label and accessible name.
+
+```html
+<div role="link" aria-label="Next Page">next page</div>
+```
+
+#### Passed example 3
+
+Full visible label is included in the accessible name.
+
+```html
+<button name="link" aria-label="Next Page in the list">Next Page</button>
+```
+
+#### Passed example 4
+
+Full visible label is included in the accessible name.
+
+```html
+<head>
+  <style>
+    .sr-only {
+      position:absolute;
+      left:-10000px;
+      top:auto;
+      width:1px;
+      height:1px;
+      overflow:hidden;}
+  </style>
+</head>
+
+<body>
+  <a href="home.html">Read more<span style="sr-only"> about our pricing</span></a>
+</body>
+```
+
+### Failed
+
+#### Failed example 1
+
+Visible label is not included in accessible name.
+
+```html
+<div role="link" aria-label="OK">Next</div>
+```
+
+#### Failed example 2
+
+Not all of visible label is included in accessible name.
+
+```html
+<button name="link" aria-label="the full">The full label</button>
+```
+
+#### Failed example 3
+
+The full visible label is split out across the accessible name.
```

Comment:
Could you please explain why this fails? The SC states that the text of the label should be at the beginning of the accessible name, but that is just a best practice. I don't interpret it as a violation if it does not occur. 

Also, the expectation for this rule does not state that the visible text should be at the beginning of the accessible name or that it cannot be split out across the accessible name. 

I tried this on Chrome with VoiceOver and the AT reads "Link, that opens a new tab Link to the homepage". So, the visible text is there, it's just not at the start of accessible name.

Reply:
It is not a matter of starting it at the beginning. This example fails because people that use voice controls might say "click link to the homepage" to their computers because that is what they can see. But there is no control with "link to the homepage" anywhere in there because there is invisible text between "link" and "to".

The word "complete" in the expectation might be ambiguous.  I would say that the complete visible text "link to the homepage" is not in there. There is "link" and there is "to the homepage", and neither are complete.

But you could also argue that "all the visible words are in the name, so the name has the complete visible text".

Maybe we should add the word "uninterrupted".

Reply:
The definition of "included in" also has this: 

> ... includes any consecutive sequence of encoded characters that [matches](#matching-characters) the former.

So the full accessible name has to be included character by character in the order that the characters are in, according to this rule right now.

Reply:
It might be a good idea to have this explicit in the comment. Something like:

The full visible label is split out across the accessible name therefore not being [included](#included-characters) in it.